docs/naps/8-telemetry.md

```diff
@@ -0,0 +1,190 @@
+4. `full` - same as in `middle` but also collects information about plugin usage by binding to app-model and logging plugin commands used. Also basic information about data like type (`np.ndarray`, `dask.array`, `zarr.Array`, etc.) and its size is collected.
+
+There should be a visible indicator that telemetry is enabled (for example on the status bar). 
+
+The second part of this work should be to setup the server to collect telemetry data. After collecting data, it should provide a basic public dashboard that will allow the community to see aggregated information.
+
+I propose to have the following data retention policy:
+
+1) Up to 2 weeks for logs.
+2) up 2 months of raw data (1 month of collection, then aggregation and time to validate aggregated data),
+3) infinite of aggregated data.
+  
+## Privacy assessment
+
+During the preparation of this NAP we assume that none of the collected data will be presented in 
+a form that allows to identify a single user or identify a research area of user. We also select a set of data that will be collected to minimize the possibility of revealing fragile data, but it is impossible to guarantee that it will not be possible to identify a single user (for example by checking installed plugin combinations).
+
+Because of this, we propose to not publish raw data and only show aggregated results. The aggregation will be performed using scripts. Napari core devs will access raw data only if there are errors in the aggregation process.
+
+We also will publish a list of endpoints for each level of telemetry, so the given level of telemetry could be blocked on the organization level (for example by the rule on the firewall).
+
+
+If someone found that we are publishing some problematic data we will remove them and update the aggregation process to prevent such a situation in the future.
+This NAP will be updated to reflect the current state of telemetry. 
+
+ 
+## Related Work 
+
+Total systems:
+https://plausible.io/
+https://sentry.io/
+https://opentelemetry.io/
+
+Visualizations:
+https://github.com/grafana/grafana
+  
+
+  
+## Implementation 
+
+The main thing for implementation should be the low cost of maintenance. So the solution should be as simple as possible. We could either use existing solutions on the server side or implement our own.
+
+The benefit of existing solutions is that most of the work is already done. The downside is that it may require additional cost of maintenance. This cost may be caused by many features that are not needed for napari and could increase the risk of leaking data.  Quick checks of their code revealed they are implemented in techniques that are not familiar to napari core devs. So, if we decide to use them, we should select an SAS solution that will be maintained by the company.
+
+
+For now, I suggest creating a simple REST API server for collecting the data. 
+It could be a simple Python FastAPI server that will store data in the SQLite database.
```

Comment:
Any communication with the server needs to be encrypted to avoid MITM. I would specify explicitly the protocol indended to be used for that, i.e. SSH/HTTPS/other